source/future.tex

```diff
@@ -98,20 +98,64 @@
 prefix \tcode{++} and \tcode{\dcr} expressions\iref{expr.pre.incr}
 of \tcode{volatile}-qualified arithmetic and pointer types are deprecated.
 
+\begin{example}
+\begin{codeblock}
+volatile int velociraptor;
+++velociraptor;                 // deprecated: is this a single memory operation, or two
+\end{codeblock}
+\end{example}
+
+
 \pnum
 Certain assignments
 where the left operand is a \tcode{volatile}-qualified non-class type
 are deprecated; see~\ref{expr.ass}.
 
+\begin{example}
+\begin{codeblock}
+int neck, tail;
+volatile int brachiosaur;
+brachiosaur = neck;                 // OK, a volatile store
+tail = brachiosaur;                 // OK, a volatile load
+tail = brachiosaur = neck;          // deprecated: does this access \tcode{brachiosaur} once or twice?
+brachiosaur += neck;                // deprecated: does this access \tcode{brachiosaur} once or twice?
+brachiosaur = brachiosaur + neck;   // OK, a volatile load, an addition, a volatile store
+\end{codeblock}
+\end{example}
+
+
 \pnum
 A function type\iref{dcl.fct}
 with a parameter with \tcode{volatile}-qualified type or
 with a \tcode{volatile}-qualified return type is deprecated.
 
+\begin{example}
+\begin{codeblock}
+volatile struct amber jurassic();  // deprecated: a volatile return type has no meaning
+
+// deprecated: volatile parameters aren't meaningful to the caller, volatile only applies within the function
+void trex(volatile short left_arm, volatile short right_arm);
+
+void fly(volatile struct pterosaur* pteranodon);   // OK, the pointer isn't volatile, the data it points to is
+\end{codeblock}
+\end{example}
+
+
 \pnum
 A structured binding\iref{dcl.struct.bind} of a \tcode{volatile}-qualified type
 is deprecated.
 
+\begin{example}
+\begin{codeblock}
+struct linhenykus { volatile short forelimb; };
+void park(linhenykus alvarezsauroid) {
+  auto [what_is_this] = alvarezsauroid;    // deprecated: does the binding copy the \tcode{forelimb}s?
```

Comment:
This example is not deprecated. The deprecated case is
```
struct whatever { int n; };
volatile auto [what_is_this] = whatever{1};
```

Reply:
Done.